hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/RandomKeyReader.java

```diff
@@ -0,0 +1,104 @@
+package org.apache.hadoop.ozone.freon;
+
+import com.codahale.metrics.Timer;
+import kotlin.Pair;
+import org.apache.commons.io.IOUtils;
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.ozone.client.OzoneBucket;
+import org.apache.hadoop.ozone.client.OzoneClient;
+import org.apache.hadoop.ozone.client.OzoneKey;
+import org.apache.hadoop.ozone.client.OzoneVolume;
+import org.apache.hadoop.ozone.client.io.OzoneInputStream;
+import picocli.CommandLine;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Random;
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Reads keys from random volume/bucket pairs.
+ */
+@CommandLine.Command(name = "random-key-reader",
+        aliases = "rkr",
+        description = "Read keys from random volume/buckets",
+        mixinStandardHelpOptions = true)
+public class RandomKeyReader extends BaseFreonGenerator
+        implements Callable<Void> {
+  @CommandLine.Option(
+          names = "--om-service-id",
+          description = "OM Service ID"
+  )
+  private String omServiceID = null;
+  private OzoneClient[] ozoneClients;
+  private int clientCount;
+  private Timer timer;
+  private AtomicInteger readKeyCount = new AtomicInteger();
+
+  public Void call() throws Exception {
```

Comment:
Ideally this method should read something like:
init()
setupClients()
gatherVolumeBucketInfo()
runTests()
teardownClients()
report()

And the respective methods should contain the client initialization, the volume-bucket mapping setup, and the close of clients, then printing the report.

Also note that closing the clients might throw an exception, with that it might be useful to catch that exception in the teardownClients() method, and just report them to the console.